x-pack/plugins/security_solution/server/lib/detection_engine/rule_management/logic/import/action_connectors/utils/index.ts

```diff
@@ -0,0 +1,129 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import { pick } from 'lodash';
+import type { SavedObjectsImportFailure } from '@kbn/core-saved-objects-common';
+import type { SavedObject } from '@kbn/core-saved-objects-server';
+import type { ActionsClient } from '@kbn/actions-plugin/server';
+import type { BulkError } from '../../../../../routes/utils';
+import { createBulkErrorObject } from '../../../../../routes/utils';
+import type { RuleToImport } from '../../../../../../../../common/detection_engine/rule_management';
+import type {
+  ActionRules,
+  ConflictError,
+  ErrorType,
+  ImportRuleActionConnectorsResult,
+  SOError,
+} from '../types';
+
+export const returnErroredImportResult = (error: ErrorType): ImportRuleActionConnectorsResult => ({
+  success: false,
+  errors: [handleActionConnectorsErrors(error)],
+  successCount: 0,
+  warnings: [],
+});
+
+export const handleActionsHaveNoConnectors = (
+  actionsIds: string[],
+  actionConnectorRules: ActionRules
+): ImportRuleActionConnectorsResult => {
+  const ruleIds: string = [...new Set(Object.values(actionConnectorRules).flat())].join();
+
+  if (actionsIds && actionsIds.length) {
+    const errors: BulkError[] = [];
+    const errorMessage =
+      actionsIds.length > 1
+        ? 'connectors are missing. Connector ids missing are:'
+        : 'connector is missing. Connector id missing is:';
+    errors.push(
+      createBulkErrorObject({
+        statusCode: 404,
+        message: `${actionsIds.length} ${errorMessage} ${actionsIds.join(', ')}`,
+        ruleId: ruleIds,
+      })
+    );
+    return {
+      success: false,
+      errors,
+      successCount: 0,
+      warnings: [],
+    };
+  }
+  return {
+    success: true,
+    errors: [],
+    successCount: 0,
+    warnings: [],
+  };
+};
+
+export const handleActionConnectorsErrors = (error: ErrorType, id?: string): BulkError => {
```

Comment:
Should error be of type `SavedObjectsImportFailure['error']` since that's what is being iterated over in `mapSOErrorToRuleError()`? Then you can use type narrowing below instead of casting 🙂 

```suggestion
export const handleActionConnectorsErrors = (error: SavedObjectsImportFailure['error'], id?: string): BulkError => {
```

Reply:
Unfortunately the ` actionsImporter.import` doesn't return always the `SavedObjectsImportFailure['error']` especially when it comes to `Server Errors`  that's why I had to add the `ErrorType` type